pkg/util/log/log.go

```diff
@@ -6,51 +6,24 @@ import (
 	kitlog "github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	dslog "github.com/grafana/dskit/log"
-	"github.com/grafana/dskit/server"
 )
 
 // Logger is a shared go-kit logger.
 // TODO: Change all components to take a non-global logger via their constructors.
 // Prefer accepting a non-global logger as an argument.
 var Logger = kitlog.NewNopLogger()
 
-// InitLogger initialises the global gokit logger and overrides the
-// default logger for the server.
-func InitLogger(cfg *server.Config) {
-	logger := kitlog.NewLogfmtLogger(kitlog.NewSyncWriter(os.Stderr))
-	if cfg.LogFormat.String() == "json" {
-		logger = kitlog.NewJSONLogger(kitlog.NewSyncWriter(os.Stderr))
-	}
+// InitLogger initialises the global gokit logger and returns that logger.
+func InitLogger(logFormat string, logLevel dslog.Level) kitlog.Logger {
+	writer := kitlog.NewSyncWriter(os.Stderr)
+	logger := dslog.NewGoKitWithWriter(logFormat, writer)
 
-	// add support for level based logging
-	logger = level.NewFilter(logger, LevelFilter(cfg.LogLevel.String()))
+	// use UTC timestamps and skip 5 stack frames.
+	logger = kitlog.With(logger, "ts", kitlog.DefaultTimestampUTC, "caller", kitlog.Caller(5))
```

Comment:
What does it mean to change the call stack depth here?  Do we now report a different line is doing the logging?

Reply:
Hi @zalegrala,
The reason for that is the implementation of `dslog.NewGoKitWithWriter()` that is now used to create the actual logger, and that replaces the “manual” creation that was used before.
I did a manual test to check the outcome and it prints tve same source and line it used to print before my change.
I will prepare a Unit test that ensures the right thing is printed once I am back from PTO. I will also replace the usage of `RateLimitedLogger` with the one from `dskit`.